### DISCUSSION TITLE
Welcome to slacker Discussions!

### DISCUSSION THREAD

**Discussion by shomali11** [shomali11/slacker#66] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MaximilianoMonaco** [comment]
Hi, I'm building a bot that responds per thread, how can I make it not respond more than once in the same thread?

**Reply by arusso** [reply]
What version of slacker? How many instances are you running? Can you share your command code?

I would suggest asking this in the #slacker-framework channel in the Gophers slack as there will be more eyes on it. This forum never really took off.

**Reply by MaximilianoMonaco** [reply]
`package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/shomali11/slacker"
)

// Define the command handler
func collieHelpCommand(analyticsChannel <-chan *slacker.CommandEvent, done chan<- bool) {
	for event := range analyticsChannel {
		fmt.Println("Command Events")
		fmt.Println(event.Timestamp)
		fmt.Println(event.Command)
		fmt.Println(event.Parameters)
		fmt.Println(event.Event)

	}
	done <- true
}

// Main function
func main() {

	collie := slacker.NewClient(os.Getenv("SLACK_BOT_TOKEN"), os.Getenv("SLACK_APP_TOKEN"))

	info := `
	Gracias por contactarte al Canal de Colaboradores!

	Para consultas y gestiones de permisos de Colaboradores puedes hacerlo directamente en nuestro <https://sx.adminml.com/collaborators/user-management|Admin>
	
	Para contexto general de la herramienta puedes consultar nuestro <https://docs.google.com/presentation/d/15LZpkxky8_edsx5wp-0Cq-UjffPj4rQz-9vXRf-rhqs/edit#slide=id.g11da6f90a4f_0_0|Guideline>
	
	Obrigado por entrar em contato com o Canal do Colaborador!
	
	Para consultas e gerenciamento de permissões de Colaborador, você pode fazê-lo diretamente em nosso <https://sx.adminml.com/collaborators/user-management|Admin>
	
	Para o contexto geral da ferramenta, você pode consultar nossa <https://docs.google.com/presentation/d/15LZpkxky8_edsx5wp-0Cq-UjffPj4rQz-9vXRf-rhqs/edit#slide=id.g11da6f90a4f_0_0|Guideline>`

	//done := make(chan bool)

	//go collieHelpCommand(collie.CommandEvents(), done)
	collie.Command("Collie", &slacker.CommandDefinition{
		Handler: func(botCTX slacker.BotContext, request slacker.Request, response slacker.ResponseWriter) {
			response.Reply("Collie is running", slacker.WithThreadReply(true))
		},
	})

	collie.DefaultCommand(func(botCtx slacker.BotContext, request slacker.Request, response slacker.ResponseWriter) {
		response.Reply(info, slacker.WithThreadReply(true))
	})

	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	err := collie.Listen(ctx)
	if err != nil {
		log.Fatal(err)
	}

}`
github.com/shomali11/slacker v1.4.1
